Apu2 diagnostic image

Asiderr opened the pull request.

Commit:
recipes-kernel/linux/linux-yocto_4.19%.bbappend: changing yocto version

Commit:
recipes-extended/xen/xen_%.bbappend: renaming bbappend file, adding symbolic link

Commit:
kas.yml: creating kas file that builds an image

Commit:
example/: removing examples

Commit:
conf/distro/pce-base.conf: removing PCENGINES_EXTRA_DISTRO_FEATURES
conf/distro/pce-xen.conf: removing PCENGINES_EXTRA_DISTRO_FEATURES and adding DISTO_FEATURES
conf/layer.conf:changing LAYERSERIES_COMPAT

Commit:
conf/distro/include/: recreating includes for newer yocto verison

Commit:
recipes-extended/images/xen-dom0-image.bb: adding dom0 recipe
recipes-extended/xen/xen_git.bbappend: specifying repo revision

Commit:
git.bbappend: Adding ipxe patch to the recipe
recipes-extended/ipxe/ipxe_git.bbappend: Fix detection of host compiler "no-pie"/"nopie" and link flags.

Commit:
README.md: the warning message update

Commit:
conf/machine/include/tune-family16h.inc: Delete the unnecessary include file
conf/machine/include/pcengines-apux.inc: Change the gcc tune

Commit:
conf/machine/include/pcengines-apux.inc: Specify distro for KERNEL_MODULE_AUTOLOAD

Commit:
kas.yml: Remove WKS_FILE and IMAGE_BOOT_FILES variable

Commit:
README.md: Add the 'Flashing drive with dd' section and 'Bootup log' section

Commit:
README.md: Add link to bootup log
bootup.log: Add boot log

Commit:
bootup.log: Update the bootlog

Commit:
recipes-extended/images/xen-dom0-image.bb: Add a seabios to the dom0 image

Commit:
conf/layer.conf: add zeus to LAYERSERIES_COMPAT
kas.yml: update refspecs

macpijan committed:
pce-distro.inc: add virtualization to default features

Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>

Commit:
kas.yml update refspeces

macpijan commented on recipes-tpm2-tss/tpm2-tss/tpm2-tss_%.bbappend:
```diff
@@ -0,0 +1,2 @@
+FILES_${PN} = "/lib/udev"
```
@Asiderr What is the status of this upstream patch?

Asiderr replied:
@macpijan Sent and stuck.

macpijan commented on recipes-extended/images/apu2-diagnostic-image.bb:
```diff
@@ -0,0 +1,15 @@
+require recipes-extended/images/docker-image-minimal.bb
+
+IMAGE_INSTALL_append = " \
+    flashrom \
```
@Asiderr I do not think your flashrom recipe is being picked up here. I believe the one from meta-openembedded is being used. You can verify that in your build dir.

macpijan commented on recipes-bsp/flashrom_git.bb:
```diff
@@ -0,0 +1,19 @@
+
+SRC_URI = "git://github.com/pcengines/flashrom.git \
+           "
+SRC_URI[md5sum] = "91bab6c072e38a493bb4eb673e4fe0d6"
```
@Asiderr This recipe will not build. When using the `git` fetcher, the SRC_URI[md5sum|sha256sum] are not used. They are used in case of downloaded files / tarballs. In case of git repo, SRCREV needs to be set: https://www.yoctoproject.org/docs/current/mega-manual/mega-manual.html#var-SRCREV

macpijan replied:
You should use https://www.yoctoproject.org/docs/current/mega-manual/mega-manual.html#var-bb-PREFERRED_PROVIDER to tell which flashrom should be used.

macpijan commented on recipes-bsp/flashrom_git.bb:
```diff
@@ -0,0 +1,19 @@
+SRC_URI[sha256sum] = "aeada9c70c22421217c669356180c0deddd0b60876e63d2224e3260b90c14e19"
+
+S = "${WORKDIR}/flashrom-v${PV}"
+
```
@Asiderr You can add custom PV to easily distinguish our custom flashrom. Something like PV = "1.0.1-pcengines+${SRCPV}" - depending on which flashrom version we are basing here.

macpijan commented on kas.yml:
```diff
@@ -4,7 +4,7 @@ header:
 machine: pcengines-apu2
 target:
-  - core-image-minimal
+  - apu2-diagnostic-image
```
@Asiderr Please create another kas file, as explained in the docker PR. Let me know if something is not clear.

macpijan commented on recipes-bsp/flashrom_git.bb:
```diff
@@ -0,0 +1,19 @@
+SRC_URI[md5sum] = "91bab6c072e38a493bb4eb673e4fe0d6"
+SRC_URI[sha256sum] = "aeada9c70c22421217c669356180c0deddd0b60876e63d2224e3260b90c14e19"
+
+S = "${WORKDIR}/flashrom-v${PV}"
```
@Asiderr You can use ${PN} instead of `flashrom` in a recipe. But it won't work here. With git fetcher, the `S = "{WORKDIR}/git"` most of the times.

Commit:
recipes-extended/images/apu2-diagnostic-image.bb: Create image including required tools

Commit:
recipes-kernel/linux/: Add kernel support for edac

Commit:
recipes-tpm2-tss/tpm2-tss/tpm2-tss_%.bbappend: Change udevrule direction

Commit:
recipes-extended/images/docker-image-minimal.bb: Create docker image

Commit:
recipes-containers/docker/docker-ce_git.bbappend: Add kernel-module

Commit:
recipes-extended/images/docker-image-minimal.bb: Replace openssh package with
IMAGE_FEATURE

Commit:
kas.yml: Create kas-base file kas-docker.yml: Create the kas config for docker image

Commit:
kas-diagnostic-image.yml: Create kas config for diagnostic image

Commit:
flashrom_git.bb: Add SRC_REV and change the SRC_URI
flashrom_git.bb: Fix recipe directory
apu2-diagnostic-image.bb: Change the flashrom provider

Signed-off-by: Norbert Kaminski <norbert.kaminski@3mdeb.com>

macpijan commented on recipes-bsp/flashrom_git.bb:
```diff
@@ -0,0 +1,19 @@
+DESCRIPTION = "flashrom is a utility for identifying, reading, writing, verifying and erasing flash chips"
```
@Asiderr  This file should be removed

Asiderr replied:
@macpijan Deleted.

macpijan commented on recipes-bsp/flashrom/flashrom_git.bb:
```diff
@@ -0,0 +1,20 @@
+
+DEPENDS = "pciutils libusb libusb-compat"
+
+PV = "1.0.1-pcengines+${SRCPV}"
```
@Asiderr Where did you set which flashrom  version shall be used?

Asiderr replied:
@macpijan I set the `PREFERRED_PROVIDER_pcengines/bsp = "flashrom"` in the apu2-diagnostic-image.bb file.

macpijan replied:
This is really weird config. Use `PREFERRED_Version_flashrom = "1.0.1-pcengines%"` and place it in distro config file (not in image recipe).

Asiderr replied:
@macpijan Changed in the last commit.

Commit:
layer.conf: Add PREFERRED_VERSION_flashrom
apu2-diagnostic-image.bb: Delete PREFERRED_PROVIDER

Signed-off-by: nkaminski <norbert.kaminski@3mdeb.com>
